packages/search-metadata-previews/tests/SnippetEditorTest.js

```diff
@@ -1,43 +1,65 @@
-// import SnippetEditor from "../src/snippet-editor/SnippetEditor";
-// import React from "react";
-// eslint-disable-next-line import/named -- this concerns a mock
-import { focus } from "@yoast/replacement-variable-editor";
+import React from "react";
+import SnippetEditor from "../src/snippet-editor/SnippetEditor";
+import { fireEvent, render, screen } from "./test-utils";
 
+const defaultData = {
+	title: "Test title",
+	slug: "test-slug",
+	description: "Test description, %%replacement_variable%%",
+};
 
-describe( "SnippetEditor", () => {
-	it( "closes when calling close()", () => {
-		focus.mockClear();
-	} );
+const defaultArgs = {
+	baseUrl: "http://example.org/",
+	siteName: "Test site name",
+	data: defaultData,
+	isOpen: true,
+	onChange: jest.fn(),
+};
 
-	it( "highlights the active ReplacementVariableEditor when calling setFieldFocus", () => {
-		focus.mockClear();
+describe( "SnippetEditor", () => {
+	it( "Mobile mode", async() => {
+		const { container } = render( <SnippetEditor { ...defaultArgs } /> );
+		const desktopRadioInput = screen.getByLabelText( "Desktop result" );
+		const mobileRadioInput = screen.getByLabelText( "Mobile result" );
+		expect( desktopRadioInput ).not.toBeChecked();
+		expect( mobileRadioInput ).toBeChecked();
+		expect( container ).toMatchSnapshot();
 	} );
+	it( "Desktop mode and should switch to mobile", async() => {
+		const { container } = render( <SnippetEditor { ...defaultArgs } mode="desktop" /> );
+		const desktopRadioInput = screen.getByLabelText( "Desktop result" );
+		const mobileRadioInput = screen.getByLabelText( "Mobile result" );
+		expect( desktopRadioInput ).toBeChecked();
+		expect( mobileRadioInput ).not.toBeChecked();
+		expect( container ).toMatchSnapshot();
 
-	it( "switches modes when changing mode switcher input", () => {
-
+		fireEvent.click( mobileRadioInput );
+		expect( defaultArgs.onChange ).toHaveBeenCalledWith( "mode", "mobile" );
 	} );
 
-	describe( "shallowCompareData", () => {
-		it( "returns false when there is no new data", () => {
-
-		} );
-
-		it( "returns true when one replacement variable has changed", () => {
-
-		} );
-
-		it( "returns true when multiple data points have changed", () => {
-		} );
+	it( "Without close snippet editor button", async() => {
+		const { container } = render( <SnippetEditor { ...defaultArgs } showCloseButton={ false } /> );
+		expect( container ).toMatchSnapshot();
```

Comment:
Maybe you can just check for the close button instead of the snapshot?

```suggestion
		render( <SnippetEditor { ...defaultArgs } showCloseButton={ false } /> );
		expect( screen.queryByText( "Edit snippet" ) ).not.toBeInTheDocument();
```